lib/improver/ensemble_calibration/ensemble_calibration.py

```diff
@@ -598,9 +598,12 @@ def convert_to_cubelist(cubes, cube_type="forecast"):
                 forecast_period_constr)
 
             # Extract truth matching the time of the historic forecast.
+            reference_time = unit.num2date(
+                historic_forecast_cube.coord("time").points,
+                historic_forecast_cube.coord("time").units.name,
+                historic_forecast_cube.coord("time").units.calendar)
```

Comment:
or add ```from improver.utilities.temporal import iris_time_to_datetime``` at the top and the reference_time line becomes:

```reference_time = iris_time_to_datetime(historic_forecast_cube.coord("time").copy())```
(or we could modify the function itself to work on a copy, which would probably be nicer in retrospect, rather than modifying the cubes own time coordinate!).

Whilst I'm on about it, the ```datetime_constraint``` function in utilities.temporal will not work with iris2 anymore given that it uses point equalities. This probably needs to be looked at as part of the iris2 upgrade work.

Reply:
Actually I think `datetime_constraint` will be OK - these unit tests don't break when run with iris 2.  This doesn't fall foul of the `cell_datetime_object=True` change as it works with datetimes anyway.  The epic #215 has details of which tests fail, and although some temporal utilities tests do this is not one of them.